### PR TITLE
Hide away imports and info selection

### PR DESCRIPTION
**Pull request by scarlehoff**
This PR hides away dependencies, imports and makes the selection of the tool not rely on all tools being available (in the sense of importable).

To be completely honest, I think we should still remove dependencies (or make more dependencies optional) since for heavy tools such as Matrix we potentially want to be able to run pinefarm just to "transpile" the pinecard into a matrix runcard, so ideally that part would run with python 3.6 even. But that would require too many changes, so for now it is >=3.9. I've added the >= 3.8 marker to nnpdf since it might be the only one stopping 3.8, but I'd say we want either to go all the way down to 3.6 or stay at 3.9. 3.7/3.8 are useless since I haven't seen any computers left in that middle ground*.

*the lcm is in 3.5 but that's too much

### CODE REVIEW

**Comment by felixhekhorn** [comment]
> To be completely honest, I think we should still remove dependencies (or make more dependencies optional)

this could be done - but I'm not sure you want to do it here and I don't know how many we can actually strip ... the only obvious candidate is `pygit2`: used to install pineappl CAPI -> should be replaced by [pre-compiled binaries](https://nnpdf.github.io/pineappl/docs/installation.html#using-pre-built-binaries) and then dropped completely.

What else did you want to move?

**Comment by scarlehoff** [comment]
Nono, I don't want to do it here since it would require testing. It was just a thought (and at the moment we won't deal with the cluster running so you can always prepare the runcards in your personal computer and then submit it to a cluster)

**Review by felixhekhorn — APPROVED** [review]

File: src/pinefarm/external/__init__.py [hunk 1]
```diff
@@ -1,55 +1,52 @@
...
 
 All imports of external interfaces are 'hidden'
-under the function ``decide_external_tool`` to avoid
-unnecessary installations of external codes.
+under the function ``decide_external_tool`` to avoid an
+eager import and thus unnecessary installations of external codes.
 """
 
 from ..configs import configs
+from .interface import External
```
Comment (scarlehoff) [inline src/pinefarm/external/__init__.py:9]: I don't think we should have this import just for the `->`

File: src/pinefarm/external/__init__.py [hunk 2]
```diff
@@ -1,3 +1,52 @@
...
+"""External interfaces module.
 
-from . import mg5, vrap, yad
+All imports of external interfaces are 'hidden'
+under the function ``decide_external_tool`` to avoid an
+eager import and thus unnecessary installations of external codes.
+"""
+
+from ..configs import configs
+from .interface import External
```
Comment (scarlehoff) [inline src/pinefarm/external/__init__.py:9]:
```suggestion
```

File: src/pinefarm/external/__init__.py [hunk 3]
```diff
@@ -1,3 +1,52 @@
...
+
+
+def decide_external_tool(dsname: str) -> tuple[External, str]:
```
Comment (scarlehoff) [inline src/pinefarm/external/__init__.py:12]:
```suggestion
def decide_external_tool(dsname: str):
```

File: src/pinefarm/external/__init__.py (hunk 1 above)
Comment (scarlehoff) [reply src/pinefarm/external/__init__.py:9]: This function should be the first entry into external, without having to worry what actually happens in `.interface`. If it were used for something is ok, but I'd rather not have an annotation here.

Comment (felixhekhorn) [reply src/pinefarm/external/__init__.py:9]: fine

**Pull request merged**